✨Add feature for: prioritization of årskull to events

solarmarius opened the pull request:
This PR implements a priority registration system that allows specific årskull (graduation year) cohorts to register for events before the general registration opens. This addresses the need to give certain year groups early access to limited capacity events.

- Implement priority registration date and time fields
- Add validation for prioritized years (comma-separated format)
- Update event models with new prioritizedRegistrationDate and prioritizedYears fields
- Show priority registration info to eligible users in EventBody component
- Add disabling of 'classof' field in user settings if a year already is set

View for eglible member
<img width="867" height="648" alt="image" src="https://github.com/user-attachments/assets/cee1911b-dc51-4d8b-b5bd-3b16e3da86ad" />

View for regular members
<img width="782" height="545" alt="image" src="https://github.com/user-attachments/assets/82b512e6-23a4-4ab4-8495-3f1a3ef880c2" />

View for admins when creating an event
<img width="1100" height="835" alt="image" src="https://github.com/user-attachments/assets/b3ea85da-694a-40b2-87b4-56190a4301f6" />

solarmarius committed:
✨add prioritization of årskull to events

- Implement priority registration date and time fields
- Add validation for prioritized years (comma-separated format)
- Update event models with new prioritizedRegistrationDate and prioritizedYears fields
- Show priority registration info to eligible users in EventBody component
- Add disabling of 'classof' field in user settings if a year already is set

TorgrimRL commented on src/components/molecules/forms/eventForm/EventForm.tsx:
```diff
@@ -199,23 +263,47 @@ const EventForm = () => {
+            minWidth={20}
+            onChange={onFieldChange}
+          />
+          <TextField
```
Suggestion to make it visualy the same as time and date when not in focus

<img width="870" height="156" alt="Image" src="https://github.com/user-attachments/assets/6d462733-99b1-4f4c-9c5f-0cda047be558" />

<img width="776" height="125" alt="Image" src="https://github.com/user-attachments/assets/5993122c-7ae3-419d-a21d-a171c50c285f" />

TorgrimRL commented on src/components/molecules/event/eventBody/EventBody.tsx:
```diff
@@ -425,21 +525,60 @@ export const EventInfo: React.FC<{ event: Event; role: RoleOptions }> = ({
+                  event.prioritizedYears?.includes(parseInt(userClassof)) ? (
+                    <div>
+                      {event.registrationOpeningDate && (
+                        <p style={{ fontWeight: 'lighter' }}>
```
Try to keep static styling effects in scss

solarmarius committed:
♻️ refactor inline styles to CSS modules and enhance TextField

- Extract inline styles to CSS modules in EventBody and EventForm
- Add placeholder prop support to TextField component
- Fix TextField label positioning with placeholder
- Improve DropdownHeader className handling
- Fix placeholder opacity visibility in text.scss

solarmarius commented:
- Extract inline styles to CSS modules in EventBody and EventForm
- Add placeholder prop support to TextField component, to allow elevated label styling when not in focus
- Improve DropdownHeader className handling to merge classes instead of replace
- Fix placeholder opacity visibility in text.scss
